lib/kubernetes/src/equality/utils.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2020 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const isResourceListEqual = <Resource>(
+  desiredList: Array<Resource> | void,
+  existingList: Array<Resource> | void,
+  isResourceEqual: (desired: Resource, existing: Resource) => boolean
+): boolean => {
+  if (!Array.isArray(desiredList) || !Array.isArray(existingList)) {
+    // TODO: why though?
```

Comment:
Because there's this [check afteward](https://github.com/opstrace/opstrace/blob/mosattler/equality-check-array/lib/kubernetes/src/equality/Pod.ts#L165) in the previous code. Maybe you can move that check to this function instead and simplify this?

Reply:
> Because there's this check afteward in the previous code.

The order of the check doesn't seem ideal to me. Maybe the `undefined` checks should come first, e.g.
```typescript
  if (desired.ports && existing.ports === undefined) {
    return false;
  }
  if (
    !isResourceListEqual(
      desired.ports,
      existing.ports,
      (desiredPort, existingPort) =>
        isContainerPortEqual(desiredPort, existingPort)
    )
  ) {
    return false;
  }
```